.github/workflows/python_bindings.yml

```diff
@@ -251,13 +269,84 @@ jobs:
       run: |
           set -x
           python -m pip install --upgrade pip
-          pip install requests packaging
+          # Note: if we choose to do ALL or nothing, remove  setuptools wheel twine (not needed unless trying to upload within this step)
+          pip install requests packaging setuptools wheel twine
           bindings_v=$(python ./python/find_pypi_tag.py --current)
           pip install -i https://test.pypi.org/simple/ openstudio==$bindings_v
+          mkdir wheel
 
     - name: Test the bindings
       shell: python
       run: |
           import openstudio
           m = openstudio.model.exampleModel()
           print(m)
+
+    - name: Download the specific wheel
+      # if: contains(github.ref, 'refs/tags')
+      uses: actions/download-artifact@v2
+      with:
+        path: ./wheel
+        name: openstudio-${{ matrix.name }}
+
+    - name: Extract the tar.gz, then upload
+      # if: contains(github.ref, 'refs/tags')
+      working-directory: ./wheel
+      env:
+        TWINE_USERNAME: __token__
+        TWINE_REPOSITORY: pypi
+        # Go to repo on github > Settings > Secrets and add it
+        TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+      shell: bash
+      run: |
+        tar -xzvf "openstudio-${{ matrix.name }}.tar.gz";
+        ls -R
+        python -m twine upload --skip-existing dist/*
+
+  upload_python_bindings_to_pypi:
+    # if: contains(github.ref, 'refs/tags')
+    needs: test_python_bindings
+    name: Upload to PyPi
+    runs-on: ubuntu-20.04
+    steps:
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install Python dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install setuptools wheel twine requests packaging
+        mkdir wheels
+
+    - name: Download ALL wheels
+      uses: actions/download-artifact@v2
+      with:
+        path: ./wheels
+
+    - name: Display structure of downloaded files
+      working-directory: ./wheels
+      run: |
+        ls -R
+
+    - name: Extract all tar.gz, then upload
+      working-directory: ./wheels
+      env:
+        TWINE_USERNAME: __token__
+        TWINE_REPOSITORY: pypi
+        # Go to repo on github > Settings > Secrets and add it
+        TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+      shell: bash
+      run: |
+        for f in **/*.tar.gz; do
+          mv $f .
+        done;
+
+        for f in *.tar.gz; do
+          tar -xzvf "$f";
+        done;
+
+        ls -R
+        python -m twine upload --skip-existing --repository pypi dist/*
```

Comment:
Download all artifacts, they all end up in the same dist/ subfolder that we upload.